stat_fem/estimation.py

```diff
@@ -45,13 +45,31 @@ def estimate_params_MAP(A, b, G, data, priors=[None, None, None], start=None, en
                           are parallelized. Optional, default value is ``COMM_SELF`` indicating
                           that forcing covariance solves are not parallelized.
     :type enemble_comm: MPI Communicator
-    :param kwargs: Additional keyword arguments to be passed to the minimization routine.
+    :param kwargs: Additional keyword arguments to be passed to either the Firedrake
+                   `LinearSolver` object or the Scipy `minimize` routine. See
+                   the corresponding manuals for more information.
     :returns: A LinearSolver object with the hyperparameters set to the MAP/MLE value. To extract
               the actual parameters, use the ``params`` attribute.
     :rtype: LinearSolver
     """
 
-    ls = LinearSolver(A, b, G, data, priors, ensemble_comm)
+    # extract kwargs for firedrake solver and minimize
+    
+    firedrake_kwargs = ["P", "solver_parameters", "nullspace",
```

Comment:
It might be nice to add a link here to where these parameters are documented in Firedrake (https://www.firedrakeproject.org/firedrake.html#firedrake.linear_solver.LinearSolver) ?

Reply:
Agreed, I plan to do a full sweep of the documentation to better explain this before finishing the next release. Will make an issue about this.